build.gradle.kts

```diff
@@ -0,0 +1,210 @@
+import net.minecrell.pluginyml.bukkit.BukkitPluginDescription
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import org.hidetake.groovy.ssh.core.Remote
+import org.hidetake.groovy.ssh.core.RunHandler
+import org.hidetake.groovy.ssh.core.Service
+import org.hidetake.groovy.ssh.session.SessionHandler
+import net.ltgt.gradle.errorprone.errorprone
+import net.ltgt.gradle.errorprone.CheckSeverity
+
+plugins {
+    java
+    `maven-publish`
+    id("com.palantir.git-version") version "0.12.2"
+    id("com.github.johnrengelman.shadow") version "7.1.2"
+    id("net.minecrell.plugin-yml.bukkit") version "0.5.1" // Generates plugin.yml
+    id("org.hidetake.ssh") version "2.10.1"
+    id("net.ltgt.errorprone") version "2.0.2"
+    id("net.ltgt.nullaway") version "1.3.0"
+    checkstyle
+    pmd
+}
+
+repositories {
+    mavenLocal()
+    maven {
+        url = uri("https://repo.papermc.io/repository/maven-public/")
+    }
+
+    maven {
+        url = uri("https://jitpack.io")
+    }
+
+    maven {
+        url = uri("https://oss.sonatype.org/content/repositories/snapshots/")
+    }
+
+    maven {
+        url = uri("https://raw.githubusercontent.com/TeamMonumenta/monumenta-redis-sync/master/mvn-repo/")
+    }
+
+    maven {
+        url = uri("https://raw.githubusercontent.com/TeamMonumenta/NBTEditor/master/mvn-repo/")
+    }
+
+    // NBT API, pulled in by CommandAPI
+    maven {
+        url = uri("https://repo.codemc.org/repository/maven-public/")
+    }
+
+    maven {
+        url = uri("https://repo.maven.apache.org/maven2/")
+    }
+}
+
+dependencies {
+    compileOnly("io.papermc.paper:paper-api:1.18.2-R0.1-SNAPSHOT")
+    implementation("dev.jorel.CommandAPI:commandapi-shade:8.7.0")
+    compileOnly("com.playmonumenta:nbteditor:4.0")
+    compileOnly("com.playmonumenta:redissync:3.0")
+    compileOnly("com.google.code.gson:gson:2.8.5")
+    errorprone("com.google.errorprone:error_prone_core:2.10.0")
+    errorprone("com.uber.nullaway:nullaway:0.9.5")
+}
+
+group = "com.playmonumenta"
+val gitVersion: groovy.lang.Closure<String> by extra
+version = gitVersion()
+description = "libraryofsouls"
+java.sourceCompatibility = JavaVersion.VERSION_17
+java.targetCompatibility = JavaVersion.VERSION_17
+
+// Configure plugin.yml generation
+bukkit {
+    load = BukkitPluginDescription.PluginLoadOrder.POSTWORLD
+    main = "com.playmonumenta.libraryofsouls.LibraryOfSouls"
+    apiVersion = "1.18"
+    name = "LibraryOfSouls"
+    authors = listOf("The Monumenta Team")
+    depend = listOf("CommandAPI", "NBTEditor")
```

Comment:
CommandAPI should not be a `depend` when shading.

Reply:
Good catch, will fix